### DISCUSSION TITLE
Docker Install

### DISCUSSION THREAD

**Discussion by rieders** [inventree/InvenTree#8315] [General]
Hello. I have installed Inventree on an Ubuntu system under Docker. It is now running with a few problems. But I cannot open the website. Unfortunately I do not know if I have set up caddy correctly. I would be happy if there could be instructions for the simplest installation. I have already installed inventree as a bare metal installation. But I cannot update the WLed plugin there. Thank you very much for your help.

**Comment by SchrodingersGat** [comment]
Did you follow these guides?

- https://docs.inventree.org/en/stable/start/docker/
- https://docs.inventree.org/en/stable/start/docker_install/

**Comment by rieders** [comment]
Hello
Many thanks for the quick response. I did the installation as described in the instructions, but somehow inventree doesn't want to start. what could it be? Is there a finished Raspberry pi image?



inventree@Docker-Inventree:~$ sudo docker compose up -d
WARN[0000] Found orphan containers ([inventree-caddy-1 inventree-inventree-server-run-a1fd76a1612c]) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up. 
[+] Running 4/5
 ✔ Container inventree-db      Running                                                                             0.0s 
 ✔ Container inventree-cache   Running                                                                             0.0s 
 ⠸ Container inventree-proxy   Starting                                                                            4.3s 
 ✔ Container inventree-server  Started                                                                             0.0s 
 ✔ Container inventree-worker  Started                                                                             0.0s 
Error response from daemon: driver failed programming external connectivity on endpoint inventree-proxy (9dcbdb9c01db30655340d47f7d9e89bcebb354dfdbce3877af91fef4d3e5a30f): failed to bind port 0.0.0.0:80/tcp: Error starting userland proxy: listen tcp4 0.0.0.0:80: bind: address already in use

**Reply by SchrodingersGat** [reply]
> Error starting userland proxy: listen tcp4 0.0.0.0:80: bind: address already in use

This would indicate that you have another process already web serving on port 80. If you navigate your browser to `http://localhost` you should be able to work out what it is

**Comment by rieders** [comment]
Hello. Caddy has started. Maybe something is set incorrectly in Caddy's config.

# Example Caddyfile for Inventree
# The following environment variables may be used:
# - INVENTREE_SITE_URL: The upstream URL of the Inventree site (default: inventree.localhost)
# - INVENTREE_SERVER: The internal URL of the Inventree container (default: http://inventree-server:8000)
#
# Note that while this file is a good starting point, it may need to be modified to suit your specific requirements


(log_common) {
	log {
		output file /var/log/caddy/{args[0]}.access.log
	}
}

(cors-headers) {
	header Allow GET,HEAD,OPTIONS
	header Access-Control-Allow-Origin *
	header Access-Control-Allow-Methods GET,HEAD,OPTIONS
	header Access-Control-Allow-Headers Authorization,Content-Type,User-Agent

	@cors_preflight{args[0]} method OPTIONS

	handle @cors_preflight{args[0]} {
		respond "" 204
	}
}

# Change the host to your domain (this will serve at inventree.localhost)
{$INVENTREE_SITE_URL:inventree.localhost} {
	import log_common inventree

	encode gzip

	request_body {
		max_size 100MB
	}

	handle_path /static/* {
		import cors-headers static

		root * /var/www/static
		file_server
	}

	handle_path /media/* {
		import cors-headers media

		root * /var/www/media
		file_server

		header Content-Disposition attachment

		forward_auth {$INVENTREE_SERVER:"http://inventree-server:8000"} {
			uri /auth/
		}
	}

	reverse_proxy {$INVENTREE_SERVER:"http://inventree-server:8000"}
}

**Comment by rieders** [comment]
Hallo. Ich habe nun unter debian 12 inventree installiert.  Es läuft auch . Wie kann ich die Artikel und lagerorte und Lieferanten vom alten System in das neue übertragen? 

Grüße André
